src/AggregateFunctions/AggregateFunctionFunnelAnalysis.cpp

```diff
@@ -0,0 +1,58 @@
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+#include <AggregateFunctions/AggregateFunctionFunnelAnalysis.h>
+#include <AggregateFunctions/Helpers.h>
+#include <AggregateFunctions/FactoryHelpers.h>
+#include <DataTypes/DataTypeDate.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypeArray.h>
+
+#include <ext/range.h>
+#include "registerAggregateFunctions.h"
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+namespace
+{
+
+template <template <typename> class Data>
+AggregateFunctionPtr createAggregateFunctionFunnelAnalysis(const std::string & name, const DataTypes & arguments, const Array & params)
+{
+    if (params.empty())
```

Comment:
If this function requires at least three parameters, why do you check only `params.empty()` instead of `params.size() < 3`?